### ISSUE TITLE
How can I get the link of a shared post in direct?

### ISSUE THREAD

**Issue by Pooya500** [subzeroid/instagrapi#1137]
Hello,

I am trying to use the Instagrapi library to retrieve the link of a shared post in direct messages. I have searched the documentation and tried a few different approaches, but I haven't been able to find a way to access the link of the shared post.

Can anyone provide guidance on how to retrieve the link of a shared post in direct using Instagrapi? Specifically, I would like to know if there is a method or attribute I can use to access the link of the post, and if there are any specific steps I need to take to retrieve the link.

Any help would be greatly appreciated. Thank you in advance!

**Comment by abundis-rmn2** [comment]
whats your response when you try to get the direct message?

**Comment by Pooya500** [comment]
When I receive a text message : 

`DirectMessage(id='30928464275545514639664041540190208', user_id=[3855970297](tel:3855970297), thread_id=340282366841710300949128307338723689965, timestamp=datetime.datetime(2023, 2, 17, 12, 3, 27, 959338, tzinfo=datetime.timezone.utc), item_type='text', is_sent_by_viewer=False, is_shh_mode=False, reactions=None, text='سلام', reply=None, link=None, animated_media=None, media=None, visual_media=None, media_share=None, reel_share=None, story_share=None, felix_share=None, clip=None, placeholder=None)`

When I receive a shared post : 

`DirectMessage(id='30928465794714136322186273351532544', user_id=[3855970297](tel:3855970297), thread_id=340282366841710300949128307338723689965, timestamp=datetime.datetime(2023, 2, 17, 12, 4, 50, 313634, tzinfo=datetime.timezone.utc), item_type='xma_clip', is_sent_by_viewer=False, is_shh_mode=False, reactions=None, text=None, reply=None, link=None, animated_media=None, media=None, visual_media=None, media_share=None, reel_share=None, story_share=None, felix_share=None, clip=None, placeholder=None)`

**Comment by SaeidB** [comment]
> Hello,
> 
> I am trying to use the Instagrapi library to retrieve the link of a shared post in direct messages. I have searched the documentation and tried a few different approaches, but I haven't been able to find a way to access the link of the shared post.
> 
> Can anyone provide guidance on how to retrieve the link of a shared post in direct using Instagrapi? Specifically, I would like to know if there is a method or attribute I can use to access the link of the post, and if there are any specific steps I need to take to retrieve the link.
> 
> Any help would be greatly appreciated. Thank you in advance!

use cl.direct_thread(thread_id)

**Comment by SaeidB** [comment]
> Hello,
> 
> I am trying to use the Instagrapi library to retrieve the link of a shared post in direct messages. I have searched the documentation and tried a few different approaches, but I haven't been able to find a way to access the link of the shared post.
> 
> Can anyone provide guidance on how to retrieve the link of a shared post in direct using Instagrapi? Specifically, I would like to know if there is a method or attribute I can use to access the link of the post, and if there are any specific steps I need to take to retrieve the link.
> 
> Any help would be greatly appreciated. Thank you in advance!

yes u can get url of the shared medias , but i dont remember the method in instagrapi , search this in instagrapi and u can find the method to do that :

`direct_v2/threads/340282366841710300949128217663192210248/get_items/`

**Comment by SaeidB** [comment]
> Hello,
> 
> I am trying to use the Instagrapi library to retrieve the link of a shared post in direct messages. I have searched the documentation and tried a few different approaches, but I haven't been able to find a way to access the link of the shared post.
> 
> Can anyone provide guidance on how to retrieve the link of a shared post in direct using Instagrapi? Specifically, I would like to know if there is a method or attribute I can use to access the link of the post, and if there are any specific steps I need to take to retrieve the link.
> 
> Any help would be greatly appreciated. Thank you in advance!

yes u can get url of the shared medias , but i dont remember the method in instagrapi , search `direct` in instagrapi and u can find the method to do that , similar to :

`direct_v2/threads/thread_id/get_items/`

**Comment by MahdiAbdi** [comment]
I have this problem too, and yet I haven't been able to find any answers.

**Comment by MahdiAbdi** [comment]
> > I have this problem too, and yet I haven't been able to find any answers.
> 
> cl.direct_thread

@SaeidB 

the problem is with DirectMessage. when the item type is "xma_media_share" or "xma_story_share" there is no media_id or media_pk and the rest of the fields related to the shared media are empty. so there's no way to access the original shared media info.

**Comment by MahdiAbdi** [comment]
@adw0rd

**Comment by adw0rd** [comment]
@MahdiAbdi submit your PR with a fix to the problem you are talking about

**Comment by ssriramagiri** [comment]
Fairly new here and struggling to get Proxyman running successfully. Any chance someone can help with adding the same attributes for 'xma_profile'? Similar issue as above but unsure what the answer from instagram for that call.

**Comment by Prabesh01** [comment]
Same here!
There seems to be no way to get url for item_type `xma_clip`.

The item_type `clip` becomes `xma_clip` in newer Instagram client api versions.
The `clip` message type used to provide reel's id at `message.clip.code`, and we could easily get reel's url with:
> f"https://instagram.com/reels/{message.clip.code}"

But the new `xma_clip` doesn't provide any url or reel's id:

```
id='redacted' user_id='redacted' thread_id=redacted timestamp=datetime.datetime(2026, 2, 7, 11, 40, 34) item_type='xma_clip' is_sent_by_viewer=False is_shh_mode=False reactions=None text=None reply=None link=None animated_media=None media=None visual_media=None media_share=None reel_share=None story_share=None felix_share=None xma_share=None clip=None placeholder=None client_context='redacted'
```

**Comment by subzeroid** [comment]
Thanks for the report and examples. This is now covered by PR #2484: https://github.com/subzeroid/instagrapi/pull/2484

When Instagram sends shared XMA items without a usable `target_url`, instagrapi now keeps the original payload at `message.raw_xma`. If a `target_url` is present, the existing normalized `message.xma_share` remains available.
